charmhelpers/contrib/network/ovs/ovsdb.py

```diff
@@ -28,74 +28,183 @@ class SimpleOVSDB(object):
+    # upstream adds things that downstream wants, and the cost of maintaining
+    # that will most likely be lower then the cost of finding the needle in
+    # the haystack whenever downstream code misspells something.
+    _tool_table_map = {
+        'ovs-vsctl': (
+            'autoattach',
+            'bridge',
+            'ct_timeout_policy',
+            'ct_zone',
+            'controller',
+            'datapath',
+            'flow_sample_collector_set',
+            'flow_table',
+            'ipfix',
+            'interface',
+            'manager',
+            'mirror',
+            'netflow',
+            'open_vswitch',
+            'port',
+            'qos',
+            'queue',
+            'ssl',
+            'sflow',
+        ),
+        'ovn-nbctl': (
+            'acl',
+            'address_set',
+            'connection',
+            'dhcp_options',
+            'dns',
+            'forwarding_group',
+            'gateway_chassis',
+            'ha_chassis',
+            'ha_chassis_group',
+            'load_balancer',
+            'load_balancer_health_check',
+            'logical_router',
+            'logical_router_policy',
+            'logical_router_port',
+            'logical_router_static_route',
+            'logical_switch',
+            'logical_switch_port',
+            'meter',
+            'meter_band',
+            'nat',
+            'nb_global',
+            'port_group',
+            'qos',
+            'ssl',
+        ),
+        'ovn-sbctl': (
+            'address_set',
+            'chassis',
+            'connection',
+            'controller_event',
+            'dhcp_options',
+            'dhcpv6_options',
+            'dns',
+            'datapath_binding',
+            'encap',
+            'gateway_chassis',
+            'ha_chassis',
+            'ha_chassis_group',
+            'igmp_group',
+            'ip_multicast',
+            'logical_flow',
+            'mac_binding',
+            'meter',
+            'meter_band',
+            'multicast_group',
+            'port_binding',
+            'port_group',
+            'rbac_permission',
+            'rbac_role',
+            'sb_global',
+            'ssl',
+            'service_monitor',
+        ),
+    }
+
+    def __init__(self, tool):
+        """SimpleOVSDB constructor.
 
         :param tool: Which tool with database commands to operate on.
                      Usually one of `ovs-vsctl`, `ovn-nbctl`, `ovn-sbctl`
         :type tool: str
-        :param table: Which table to operate on
-        :type table: str
         """
-        if tool not in ('ovs-vsctl', 'ovn-nbctl', 'ovn-sbctl'):
+        if tool not in self._tool_table_map:
             raise RuntimeError(
-                "tool must be one of 'ovs-vsctl', 'ovn-nbctl', 'ovn-sbctl'")
+                'tool must be one of "{}"'.format(self._tool_table_map.keys()))
         self.tool = tool
```

Comment:
I'm a bit confused as to why there is a `_tool` and a `tool`?

Reply:
Thank you for catching that :+1: